src/states/elder/mod.rs

```diff
@@ -1414,9 +1419,9 @@ impl Base for Elder {
                     signed_msg.routing_message(),
                     target
                 );
-                let p2p_node = p2p_node.clone();
+                let conn_info = p2p_node.connection_info().clone();
```

Comment:
is this clone really necessary?

Reply:
Otherwise the mutable borrow of `self` in `send_direct_message` conflicts with the immutable borrow of `self` when getting the reference to the `p2p_node` above